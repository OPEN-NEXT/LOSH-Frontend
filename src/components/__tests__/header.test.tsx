import * as React from "react";
import ShallowRenderer from "react-test-renderer/shallow"; // ES6
import Header from "../header";
import site from "../../../gatsby-config";

const renderer = ShallowRenderer.createRenderer();

jest.mock("antd", () => {
	const Menu = (): null => null;
	Menu.Item = (): null => null;

	return {
		Layout: {
			Header: (): null => null
		},
		Row: (): null => null,
		Col: (): null => null,
		Menu,
		Typography: {
			Text: (): null => null
		}
	};
});

jest.mock("../../hooks/useWindowSize", () => {
	const useWindowSize = () => ({
		width: 1230,
		height: 0
	});
	return useWindowSize;
});

const pagePadding = {
	paddingLeft: "1rem",
	paddingRight: "1rem"
};

describe("Header", () => {
	it("renders correctly", () => {
		const tree = renderer.render(
			<Header
				headerLinks={site.siteMetadata.headerLinks}
				pagePadding={pagePadding}
			/>
		);
		expect(tree).toMatchSnapshot();
	});
});
