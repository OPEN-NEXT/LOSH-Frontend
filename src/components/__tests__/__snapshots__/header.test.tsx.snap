// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<Header>
  <Row
    align="middle"
    justify="space-between"
  >
    <Col>
      <mockConstructor
        to="/"
      >
        <Text
          strong={true}
          style={
            Object {
              "color": "white",
            }
          }
        >
          Library of Open Source Hardware
        </Text>
      </mockConstructor>
    </Col>
    <Col>
      <Menu
        mode="horizontal"
      >
        <Unknown>
          <mockConstructor
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/"
          >
            <Text>
              Explore OSH Data
            </Text>
          </mockConstructor>
        </Unknown>
        <Unknown>
          <mockConstructor
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/submission"
          >
            <Text>
              Submit OSH Data
            </Text>
          </mockConstructor>
        </Unknown>
        <Unknown>
          <mockConstructor
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/ecosystem"
          >
            <Text>
              Ecosystem
            </Text>
          </mockConstructor>
        </Unknown>
        <Unknown>
          <mockConstructor
            style={
              Object {
                "textDecoration": "none",
              }
            }
            to="/get-involved"
          >
            <Text>
              Get involved
            </Text>
          </mockConstructor>
        </Unknown>
      </Menu>
    </Col>
  </Row>
</Header>
`;
